internal/controller/etcdcluster_controller.go

```diff
@@ -144,6 +144,10 @@ func (r *EtcdClusterReconciler) ensureClusterObjects(
 	if err := r.ensureClusterStatefulSet(ctx, cluster); err != nil {
 		return err
 	}
+	// 3. create or update ClusterIP Service
+	if err := r.ensureClusterClientService(ctx, cluster); err != nil {
+		return err
```

Comment:
Do I get it right that return err will make operator exit? Or will it just rise log message with error?

Reply:
If error is returned, reconciler will exit `reconcile` method with error and set CR condition to type InitializationStarted, status false